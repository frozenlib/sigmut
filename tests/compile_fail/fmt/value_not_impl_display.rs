use sigmut::{core::Runtime, signal_format};

fn main() {
    let mut rt = Runtime::new();
    struct NoDisplay;
    let _s = signal_format!("{}", NoDisplay);
}
