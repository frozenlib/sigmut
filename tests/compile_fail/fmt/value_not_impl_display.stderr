error[E0277]: `NoDisplay` doesn't implement `std::fmt::Display`
 --> tests/compile_fail/fmt/value_not_impl_display.rs:6:14
  |
6 |     let _s = signal_format!("{}", NoDisplay);
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `NoDisplay` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `NoDisplay`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = help: the trait `std::fmt::Display` is implemented for `FmtRef<'_, T>`
  = note: required for `FmtRef<'_, NoDisplay>` to implement `std::fmt::Display`
  = note: this error originates in the macro `$crate::format_args` which comes from the expansion of the macro `signal_format` (in Nightly builds, run with -Z macro-backtrace for more info)
